python/ray/serve/deployment_state.py

```diff
@@ -88,6 +89,34 @@ def print_verbose_scaling_log():
     logger.error(f"Scaling information\n{json.dumps(debug_info, indent=2)}")
 
 
+def rank_replicas_for_stopping(
+    all_avaiable_replicas: List["DeploymentReplica"],
+) -> List["DeploymentReplica"]:
+    """Prioritize replicas that have fewest copies on a node.
+
+    This algorithm helps to scale down more intelligently because it can
+    relinquish node faster. Note that this algorithm doesn't consider other
+    deployments or other actors on the same node. See more at
+    https://github.com/ray-project/ray/issues/20599.
+    """
+    # Categorize replicas to node they belong to.
+    node_to_replicas = defaultdict(list)
+    for replica in all_avaiable_replicas:
+        node_to_replicas[replica._actor._node_id].append(replica)
```

Comment:
we should prioritize replicas with `None` node_id first because they haven't been placed at all.

Reply:
also, you're using private fields here please refactor to make public

Reply:
let's make it a public interface and explicitly mention that it'll be `None` if not placed.